### ISSUE TITLE
Dispersion modelling errors

### ISSUE THREAD

**Issue by kuypes** [rich-iannone/splitr#39]
Dear Richard, 

Thank you for your work in generating and maintaining the SplitR package - it is an impressively useful toolkit. 

Please can you help me. I am trying to run a dispersion model but keep running into a problem and I have no idea what I’m doing wrong and therefore where to start to fix it. 

I have taken your example code from the Github page and changed only the location to the point source I’m trying to model: 

# Create the `dispersion_model` object, add
# a grid of starting locations, add run
# parameters, and then execute the model run
library(SplitR)
library(magrittr)

setwd("~/Documents/SplitR_wd")

# Create the `dispersion_model` object, add
# a grid of starting locations, add run
# parameters, and then execute the model run
dispersion_model <-
  create_disp_model() %>%
  add_emissions(
    rate = 5,
    duration = 6,
    start_day = "2015-07-01",
    start_hour = 0) %>%
  add_species(
    pdiam = 1,
    density = 1,
    shape_factor = 1) %>%
  add_grid(
    range = c(0.5, 0.5),
    division = c(0.1, 0.1)) %>%
  add_params(
    lat = -33.838,
    lon = 18.525,
    height = 50,
    duration = 24,
    start_day = "2017-07-01",
    start_hour = 0,
    direction = "forward",
    met_type = "gdas1",
    vert_motion = 1) %>%
  run_model

The gdas1 data has been downloaded - the first time the code ran it did that. 
Then the script begins repeatedly reporting: 

WARNING: map background file not found 
 ../graphics/arlmap    

However this file is most definitely there. I even went as far as to reinstall hysplit to make sure. I am also ok with the baseman being missing and having to build up my own. 
The real problem comes at the end when the code reports: 

Complete Parhplot
Error in read.table(file = file, header = header, sep = sep, quote = quote,  : 
  no lines available in input

I do not understand the error nor what I have done wrong. 

Furthermore, can the dispersion model be run for multiple days like the trajectories? How does one do this please? Say every 3 hours form date X to date Y?

**Comment by rich-iannone** [comment]
Things have been changing in the splitr API for dispersion modeling. I should have made this a branch (to be merged in later) so that it wouldn't be in this in-between state. I'll resolve to get things working reasonably well in `master` and going ahead with more API changes in a branch.

Anyway, this example will run a dispersion model for a single source:

```r
library(splitr)
library(tidyverse)
library(here)

dispersion_model <-
  create_dispersion_model() %>%
  add_source(
    name = "particle",
    lat = 49.0, lon = -123.0, height = 50,
    pdiam = 15, density = 1.5, shape_factor = 0.8,
    rate = 10,
    release_start = lubridate::ymd_hm("2015-07-01 00:00"),
    release_end = lubridate::ymd_hm("2015-07-01 00:00") + 
      lubridate::hours(2)
  ) %>%
  add_dispersion_params(
    start_time = lubridate::ymd_hm("2015-07-01 00:00"),
    end_time = lubridate::ymd_hm("2015-07-01 00:00") + 
      lubridate::hours(6),
    direction = "forward", 
    met_type = "reanalysis",
    met_dir = here::here("met"),
    exec_dir = here::here("out")
  ) %>%
  run_model()
```

We can have multiple `add_source()` calls. Each one will add a different source. The resulting data table has particle positions for each hour. It is available in the `dispersion_model` object.

Currently there is no way to specify repeating runs though I'd very much like to build in that functionality.

Try running this (to make sure it works on your system) and let me know your thoughts on it!

**Comment by kuypes** [comment]
Hi Rich, 

Thanks that works for the model side things! 
Also cleans up working directory - not having met files and output files in the same folder. It was something I thought to say, but forgot in my last email. 

The model runs, but am now struggling to plot the output. 
I have tried: 
# Get a data frame containing the model results
dispersion_df <-
  dispersion_model %>% get_output_tbl
# Plot
library(magrittr)
dispersion_model %>% dispersion_plot

I get: Error in unique(disp_df$hour) : object 'disp_df' not found. 

I really need to learn to code better. 
These things happen - re: branching the changes. We all learn. 

Thanks man

Brett 

——————————————

Dr. Brett Kuyper
Department of Chemistry 
University of the Western Cape 
Bellville 
Tel: (+27) 721984401

**Comment by rich-iannone** [comment]
@kuypes I just fixed `dispersion_plot()` on `master` so reinstall the package and try out

```r
dispersion_model %>% dispersion_plot()
```

once again. The map layers are pretty simplistic at this point but they will be improved.

**Comment by kuypes** [comment]
@rich-iannone Thanks for the update. That works well. 

I should have some time in the next couple of weeks to start chatting more about other aspects.

**Comment by adoubletn** [comment]
Hi Rich!

Sorry for bringing this thread up again, I am just confused. I copy/pasted exactly what was written in your comments, and following through the script R returns an error that says "There is no data available for plotting." I also noticed that when I examine the dispersion_model data in the environment tab, it returns just a list of variables. Has there been a significant change to the underlying code or am I missing a step?

I also tried hysplit_dispersion(dispersion_model)  if I needed to add this step to run the model, but then I get an error that states "The coordinate vectors are not the same length."

Thank you in advance for your time and help.
